Crashes when making a migration from legacy to new format

```
Caused by: java.util.NoSuchElementException: 
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.capitalizeWords (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.firstOfType (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.getAge (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.getInformationArray (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.getKoin (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.node (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.stringListOfType (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.getAlignment (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.loadLegacyNpcs (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.replaceQuote (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.stringOfType (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.capitalizeWords (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.firstOfType (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.getAge (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.getInformationArray (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.getKoin (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.node (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.stringListOfType (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.getAlignment (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.loadLegacyNpcs (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.replaceQuote (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcRepository.stringOfType (LegacyNpcRepository.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcImporter.importAll (LegacyNpcImporter.java)
 
  at me.kerooker.rpgnpcgenerator.legacy.repository.LegacyNpcImporter.toNpcEntity (LegacyNpcImporter.java)
 
  at me.kerooker.rpgnpcgenerator.RpgNpcGeneratorApplication.onCreate (RpgNpcGeneratorApplication.java)
 
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1026)
 
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:5099)
 
  at android.app.ActivityThread.access$1600 (ActivityThread.java:150)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1464)
 
  at android.os.Handler.dispatchMessage (Handler.java:102)
 
  at android.os.Looper.loop (Looper.java:168)
 
  at android.app.ActivityThread.main (ActivityThread.java:5885)
 
  at java.lang.reflect.Method.invoke (Method.java)
 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:797)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:687)
```

--- a/app/src/main/java/me/kerooker/rpgnpcgenerator/legacy/repository/LegacyNpcRepository.kt
+++ b/app/src/main/java/me/kerooker/rpgnpcgenerator/legacy/repository/LegacyNpcRepository.kt
@@ -18,23 +18,27 @@ class LegacyNpcRepository : KoinComponent {
 
     fun loadLegacyNpcs(): List<LegacyNpc> {
         val npcJsons = getNpcJsons()
-
-        return npcJsons.map {
-            val information = it.getInformationArray()
-            LegacyNpc(
-                information.name,
-                information.age,
-                information.race,
-                information.subRace,
-                information.gender,
-                information.alignment,
-                information.sexuality,
-                information.profession,
-                information.motivation,
-                information.personalityTraits,
-                information.fear,
-                information.languages
-            )
+    
+        return npcJsons.mapNotNull {
+            try {
+                val information = it.getInformationArray()
+                LegacyNpc(
+                    tryOrEmpty { information.name },
+                    tryOrEmpty { information.age },
+                    tryOrEmpty { information.race },
+                    tryOrEmptyN { information.subRace },
+                    tryOrEmpty { information.gender },
+                    tryOrEmpty { information.alignment },
+                    tryOrEmpty { information.sexuality },
+                    tryOrEmpty { information.profession },
+                    tryOrEmpty { information.motivation },
+                    tryOrEmptyL { information.personalityTraits },
+                    tryOrEmpty { information.fear },
+                    tryOrEmptyL { information.languages }
+                )
+            } catch (_: Exception) {
+                null
+            }
         }
     }
 
@@ -101,8 +105,26 @@ class LegacyNpcRepository : KoinComponent {
     private fun JsonObject.node(key: String) = get(key) as JsonObject
 
     private fun String.withUnderscoresReplaced() = replace("_", " ")
-
+    
     private fun String.capitalizeWords() = split(" ").joinToString(separator = " ") { it.capitalize() }
+    
+    private fun tryOrEmpty(block: () -> String) = try {
+        block()
+    } catch (_: Exception) {
+        ""
+    }
+    
+    private fun tryOrEmptyN(block: () -> String?) = try {
+        block()
+    } catch (_: Exception) {
+        null
+    }
+    
+    private fun tryOrEmptyL(block: () -> List<String>) = try {
+        block()
+    } catch (_: Exception) {
+        emptyList<String>()
+    }
 }
 
 data class LegacyNpc(

